packages/config-helpers/src/define-config.js

```diff
@@ -0,0 +1,380 @@
+		return config.name;
+	}
+
+	return `UserConfig${indexPath}`;
+}
+
+/**
+ * Gets the name of an extension.
+ * @param {SimpleExtendsElement} extension The extension.
+ * @param {string} indexPath The index of the extension.
+ * @return {string} The name of the extension.
+ */
+function getExtensionName(extension, indexPath) {
+	if (typeof extension === "string") {
+		return extension;
+	}
+
+	if (extension.name) {
+		return extension.name;
+	}
+
+	return `ExtendedConfig${indexPath}`;
+}
+
+/**
+ * Determines if a config object is a legacy config.
+ * @param {Config|LegacyConfig} config The config object to check.
+ * @return {config is LegacyConfig} `true` if the config object is a legacy config.
+ */
+function isLegacyConfig(config) {
+	for (const key of eslintrcKeys) {
+		if (key in config) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Normalizes the plugin config by replacing the namespace with the plugin namespace.
+ * @param {string} userNamespace The namespace of the plugin.
+ * @param {Plugin} plugin The plugin config object.
+ * @param {Config} config The config object to normalize.
+ * @return {Config} The normalized config object.
+ */
+function normalizePluginConfig(userNamespace, plugin, config) {
+	// @ts-ignore -- ESLint types aren't updated yet
+	const pluginNamespace = plugin.meta?.namespace;
+
+	// don't do anything if the plugin doesn't have a namespace or rules
+	if (!pluginNamespace || (!config.rules && !config.processor)) {
+		return config;
+	}
+
+	// update the rules
+
+	const result = { ...config };
+
+	if (result.rules) {
+		const ruleIds = Object.keys(result.rules);
+
+		/** @type {Record<string,RuleEntry|undefined>} */
+		const newRules = {};
+
+		for (let i = 0; i < ruleIds.length; i++) {
+			const ruleId = ruleIds[i];
+			const firstSlashIndex = ruleId.indexOf("/");
+
+			if (firstSlashIndex === -1) {
+				continue;
+			}
+
+			const ruleNamespace = ruleId.slice(0, firstSlashIndex);
+			const ruleName = ruleId.slice(firstSlashIndex + 1);
+
+			if (ruleNamespace === pluginNamespace) {
+				newRules[`${userNamespace}/${ruleName}`] = result.rules[ruleId];
+			}
+		}
+
+		result.rules = newRules;
+	}
+
+	// update the processor
+
+	if (typeof result.processor === "string") {
+		const firstSlashIndex = result.processor.indexOf("/");
+
+		if (firstSlashIndex !== -1) {
+			const ruleNamespace = result.processor.slice(0, firstSlashIndex);
+			const ruleName = result.processor.slice(firstSlashIndex + 1);
+
+			if (ruleNamespace === pluginNamespace) {
+				result.processor = `${userNamespace}/${ruleName}`;
+			}
+		}
+	}
+
+	return result;
+}
+
+/**
+ * Finds a plugin config by name in the given config.
+ * @param {Config} config The config object.
+ * @param {string} pluginConfigName The name of the plugin config.
+ * @return {Config|Config[]} The plugin config.
+ */
+function findPluginConfig(config, pluginConfigName) {
+	const [userPluginNamespace, configName] = pluginConfigName.split("/");
```

Comment:
Some plugins have configs with slashes in the config name. For example, `eslint-plugin-n` has `flat/recommended-module` config.

The following config fails with `Plugin config "n/flat/recommended-module" not found` although the config does exist, because `findPluginConfig()` looks for a config named `"flat"` instead of `"flat/recommended-module"`.

```js
// eslint.config.js
import { defineConfig } from "@eslint/config-helpers";
import n from "eslint-plugin-n";

export default defineConfig({
    plugins: {
        n
    },
    extends: ["n/flat/recommended-module"]
});
```

So I think this code should split `pluginConfigName` only by the first slash, not all slashes.